rdmo/projects/viewsets.py

```diff
@@ -877,21 +879,51 @@ def search(self, request):
         except (ValueError, TypeError):
             pass
 
-        if is_truthy(request.GET.get('collection')) and connection.vendor not in ['sqlite']:
+        if is_truthy(request.GET.get('collection')):
             # if collection is set (for checkboxes), we first select each distinct set and create a Q object with it
             # by doing so we can select an undetermined number of values which belong to an exact number of sets
             # given by settings.PROJECT_VALUES_SEARCH_LIMIT
             #
             # DISTINCT ON is not available on sqlite so we just apply the limit like for all other questions, this
             # will display the last set of checked values incomplete
-            fields = ('project_id', 'snapshot_id', 'attribute_id', 'set_prefix', 'set_index')
-            values_list = queryset.order_by(*fields) \
-                                  .distinct(*fields) \
-                                  .values(*fields) \
-                                  [:settings.PROJECT_VALUES_SEARCH_LIMIT]
+            fields = ('project_id', 'snapshot_id', 'attribute_id', 'set_prefix', 'set_index', 'text')
+
+            if connection.vendor == 'postgresql':
+                queryset = queryset.order_by(*fields).distinct(*fields)
+
+            elif connection.vendor == 'mysql':
+                queryset = queryset.annotate(
+                                row_number=Window(
+                                    expression=RowNumber(),
+                                    partition_by=[F(field) for field in fields],
+                                    order_by=F('id').asc()
+                                )
+                            ).filter(row_number=1)
+            else:
+                # Fetch the data as tuples: (id, project_id, snapshot_id, attribute_id, set_prefix, set_index)
```

Comment:
I'm trying to do the same query via python functions in here, but we don't have tests for this (eg for a sqlite db) so maybe it should be left out like before.